apiserver/pkg/util/cluster_test.go

```diff
@@ -599,6 +600,96 @@ func TestBuildHeadPodTemplate(t *testing.T) {
 	}
 }
 
+func TestNewComputeTemplate(t *testing.T) {
+	configMap, err := NewComputeTemplate(&templateWorker)
+	if err != nil {
+		t.Errorf("failed to build compute template: %v", err)
+	}
+
+	assert.Equal(t, "", configMap.Data["name"])
+	assert.Equal(t, "", configMap.Data["namespace"])
+	assert.Equal(t, "2", configMap.Data["cpu"])
+	assert.Equal(t, "8", configMap.Data["memory"])
+	assert.Equal(t, "4", configMap.Data["gpu"])
+
+	var ext map[string]uint32
+	err = json.Unmarshal([]byte(configMap.Data["extended_resources"]), &ext)
+	if err != nil {
+		t.Errorf("failed to unmarshall ExtendedResources: %v", err)
+	}
+	assert.Equal(t, uint32(32), ext["vpc.amazonaws.com/efa"])
+
+	var tolerations []*api.PodToleration
+	err = json.Unmarshal([]byte(configMap.Data["tolerations"]), &tolerations)
+	if err != nil {
+		t.Errorf("failed to unmarshall tolerations: %v", err)
+	}
+	assert.Equal(t, expectedToleration.Key, tolerations[0].Key)
+	assert.Equal(t, string(expectedToleration.Operator), tolerations[0].Operator)
+	assert.Equal(t, string(expectedToleration.Effect), tolerations[0].Effect)
+}
+
+func TestGetNodeHostIP(t *testing.T) {
+	internalIP := "10.0.0.1"
+	externalIP := "12.34.56.78"
+	invalidIP := "invalid-ip-address"
+
+	tests := []struct {
+		name        string
+		addresses   []corev1.NodeAddress
+		expectIP    string
+		expectError string
```

Comment:
nit: add a comment, like if string empty we don't expect error.
Also, usually happy paths and error paths are tested separately.

Just FYI (no change requested), If you want to save some typing, you could use test suite via `github.com/stretchr/testify/suite`.